pkg/wireguard/agent/agent.go

```diff
@@ -134,6 +146,12 @@ func (a *Agent) Init() error {
 		return err
 	}
 
+	linkMTU := mtuConfig.GetDeviceMTU() - wireguardOverhead
+	if err := netlink.LinkSetMTU(link, linkMTU); err != nil {
```

Comment:
I think this is not enough. We need to make sure that route MTU in pod netns is set accordingly (MTU minus the WG overhead), grep for `GetRouteMTU`. Maybe this is exactly the same what you went in:

> Should pkg/mtu be refactored so that the overhead computation works for both IPsec and Wireguard?

Reply:
Yep you're right. The route MTU was set too high. Modified `pkg/mtu` to handle the wireguard case. That package needs a rewrite soon, but as discussed with @brb this will be addressed in v1.11 anyway, so not worth addressing that now.